cpr/connection_pool.cpp

```diff
@@ -0,0 +1,39 @@
+#include "cpr/connection_pool.h"
+#include <curl/curl.h>
+#include <memory>
+#include <mutex>
+
+namespace cpr {
+ConnectionPool::ConnectionPool() {
+    auto* curl_share = curl_share_init();
+    this->connection_mutex_ = std::make_shared<std::mutex>();
+    
+    auto lock_f = +[](__attribute__((unused)) CURL* handle, __attribute__((unused)) curl_lock_data data, __attribute__((unused)) curl_lock_access access, void* userptr) {
+        std::mutex* lock = static_cast<std::mutex*>(userptr);
+        lock->lock();
+    };
```

Comment:
```suggestion
    auto lock_f = +[](CURL* /*handle*/, curl_lock_data /*data*/, curl_lock_access /*access*/, void* userptr) {
        std::mutex* lock = static_cast<std::mutex*>(userptr);
        lock->lock();
    };
```

I prefer just commenting them out.